### ISSUE TITLE
3+ elements for "editor.language.colorizedBracketPairs"

### ISSUE THREAD

**Issue by jacobwatkinsgit** [microsoft/vscode#170369]
Three plus keyword combinations are common for some languages, the one in mind is Lua with if-then-end. Someone had a section: `"[lua]": {
    "editor.language.colorizedBracketPairs": [
        ["function", "end"],
        ["then", "end"],
        ["do", "end"],
        ["[", "]"],
        ["{", "}"],
        ["(", ")"]
    ]
}`

This had me thinking: "why not ["if", "then", "else", "end"], ["if", "then", "end"] ? (the former would need to match before the latter) I wouldn't ask for repetitions / regex matching e.g. for "elseif"s, but a bracket colorization is *almost* a block colorization, where instead of "just for one and two", the logic goes "for one, two, and any more in the array"

**Comment by jacobwatkinsgit** [comment]
An alternative was mentioned: allowing starts and ends to be re-used to indicate indentation level. This is their example which to them makes sense for re-using colors, and 'allowing' repetitions:
```
"[lua]": {
    "editor.language.colorizedBracketPairs": [
        ["function", "end"],
        ["then", "end"],
        ["then", "elseif"],
        ["then", "else"],
        ["else", "end"]
        ["do", "end"],
        ["[", "]"],
        ["{", "}"],
        ["(", ")"],
    ],
```

**Comment by dimateos** [comment]
I agree, also mentioned in https://github.com/microsoft/vscode/issues/159586 related to C/C++.

I would keep the opening and closing words and add a list of keywords to highlight with the same color when found inside. Probably making `"editor.language.colorizedBracketPairs"` a list of objects instead of pairs, e.g.:

```jsonc
    "editor.language.colorizedBracketPairs": [
        { "bracketPair": ["#ifdef", "#endif"], "colorizedToo": ["#else"] }
    ],
```
